docs/proposals/config-dump.md

```diff
@@ -0,0 +1,248 @@
+    * `/api/alpha/configurations` 
+
+\* Query parameter id will optionally allow you to select a particular component in the case where there may be a set of them, such as in the case of kubelet. Where id is left out and we have a set of components we should expect to produce a list of configs results. 
+
+## Motivation
+
+By necessity Kubernetes is a stateful thing made up of components who have been configured in a particular way. Currently the K8s code base, as 09/2016, does not support a way nor means by which to query for the set of configurations that led to the current state of any component. As a consequence it is extremely difficult to debug the current state of a K8s cluster, especially when original configurations have been lost, or are simply not available. 
+
+## Design
+
+### API changes
+New endpoints in the api are to be created: 
+
+```
+/api/alpha/component/name/<componentName>/configurations?id=<componentId>
+```
+
+supporting the following HTTP method
+
+``` 
+GET - retrieve component configurations. **
+```
+
+\*\* The `PUT` method is to be discussed in the [Future work](#Future work) section of this proposal.
+
+An interface should be created at the component level that allows different classifications of configurations to be added to a particular component. The point being that the final configuration object is composed of a multitude classes of very specific classifications presumably defined by interested parties.
+
+Example request of a cluster with one running apiserver component:
+
+```
+kubectl get componentconfigs apiserver
+```
+
+for which the api call looks like:
+
+```
+/api/alpha/component/name/apiserver/configurations
+```
+
+both with the example output:
+
+``` 
+{
+    "kind": "component-configs",
+    "apiVersion": "alpha",
+    "metadata":{
+        "component": "apiserver",
+        "id": "718G918A-D053-EA22-BC8B-EIA8363BA071"
+        "timeStamp": "2016-09-13T14:47:35+00:00",
+        "error": []
+    },
+    "configurations": {
+        "flags": {
+            "cloud-provider": "gce",
+            ...,
+        },
+        "system": {
+            "" : ""
+        }
+    }
+}
+```
+
+We added error to metadata for the component-configs object in the case we fail to receive any configurations from a particular component. We propose that errors would look like:
+
+```
+"error": [
+    {
+        "statusCode": 408,
+        "configurationName": "flags"
+        "message": "timed out"
+    }
+]
+```
+ 
+where the message and status code represent the error received at the time of request failure. We would like that a failure to retrieve a particular configurationName would result on its own failure but not block the retrieval of those that succeeded, this probably could be the focus of a future implementation and may be beyond our current scope.
+
+### Component API changes
+
+Some components (kubelet) have endpoints that can be altered to include an endpoint such as
+
+```
+ /api/alpha/configurations
```

Comment:
We are not suggesting that configurations are blatantly exposed to be retrieved. Instead that any configuration that is to be exposed be written in such a way that it does NOT contain any sensitive data. The idea is that we do not nakedly expose some object that contains configurations, but instead that one is created is safe to view.